lib/galaxy/managers/users.py

```diff
@@ -160,6 +161,23 @@ def delete(self, user, flush=True):
                 "The configuration of this Galaxy instance does not allow admins to delete users."
             )
         super().delete(user, flush=flush)
+        self._stop_all_jobs_from_user(user)
+
+    def _stop_all_jobs_from_user(self, user):
+        active_jobs = self._get_all_active_jobs_from_user(user)
+        message = "User deleted"
+        session = self.session()
+        for job in active_jobs:
+            job.mark_deleted(self.app.config.track_jobs_in_database)
+            with transaction(session):
+                session.commit()
+            self.app.job_manager.stop(job, message)
```

Comment:
I would let the job queue handle this, which means you can also just commit once